### PR TITLE
jdbc: task2

### PR DESCRIPTION
**Pull request by slavidichkov**
I'm still not ready with the last query

### CODE REVIEW

File: src/com/clouway/task2/domain/PersonRepository.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.clouway.task2.domain;
+
+import java.util.List;
+
+/**
+ * @author Slavi Dichkov (slavidichkof@gmail.com)
+ */
+public interface PersonRepository {
+    void register(int egn, String name, int age, String email);
+    void updatePerson(int egn, String newName, int newAge, String newEmail);
```
Comment (tsony-tsonev) [inline src/com/clouway/task2/domain/PersonRepository.java]: why not pass Person object , that way you'll have 
```java
    register(person)....
    update(person)....
```

later if you want to update certain fields you can add other methods with proper names and the needed fields.

File: src/com/clouway/task2/domain/Person.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package com.clouway.task2.domain;
+
+/**
+ * @author Slavi Dichkov (slavidichkof@gmail.com)
+ */
+public class Person {
+    private final int egn;
+    private final String name;
+    private final int age;
+    private final String email;
```
Comment (tsony-tsonev) [inline src/com/clouway/task2/domain/Person.java]: you can make these filed public final and remove the getter methods

File: src/com/clouway/task2/domain/Trip.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+import java.util.Date;
+
+/**
+ * @author Slavi Dichkov (slavidichkof@gmail.com)
+ */
+public class Trip {
+    private final String city;
+    private final Date dateFrom;
+    private final Date dateTo;
```
Comment (tsony-tsonev) [inline src/com/clouway/task2/domain/Trip.java]: public final

File: src/com/clouway/task2/adapter/PersistentPersonRepository.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+    public PersistentPersonRepository(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    public void register(int egn, String name, int age, String email){
+        Connection connection = null;
+        try {
+            connection = dataSource.getConnection();
+            PreparedStatement statement=connection.prepareStatement("insert into people(EGN,age,name,email) values(?,?,?,?)");
```
Comment (tsony-tsonev) [inline src/com/clouway/task2/adapter/PersistentPersonRepository.java]: For a little improvement you can use some internal object containing the strings for the column names, cause now they are hard-coded in every request and if you want to change some column name you'll have to go through all the requests and change that column name.

File: src/com/clouway/task2/domain/TripRepository.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import java.util.List;
+
+/**
+ * @author Slavi Dichkov (slavidichkof@gmail.com)
+ */
+public interface TripRepository {
+    void register(TripRequest tripRequest);
+    List<Person> getPeopleInCityOnDate(String city, Date from, Date to);
+    List<String> mostVisitedCities();
+    void deleteRepositoryContent();
```
Comment (tsony-tsonev) [inline src/com/clouway/task2/domain/TripRepository.java]: would be better something like deleteTrips

File: test/com/clouway/jdbc/task2/PersistentPersonRepositoryTest.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    @Before
+    public void cleanUp() {
+        dataSource.setURL("jdbc:mysql://localhost:3306/task2");
+        dataSource.setUser("root");
+        dataSource.setPassword("clouway.com");
+        new DatabaseCleaner(dataSource,"people").cleanUp();
+    }
+
+    @Test
+    public void emptyRepository() {
```
Comment (tsony-tsonev) [inline test/com/clouway/jdbc/task2/PersistentPersonRepositoryTest.java]: getAllPeopleFromEmptyRepository

File: test/com/clouway/jdbc/task2/PersistentPersonRepositoryTest.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        PersistentPersonRepository userRepository = null;
+        userRepository = new PersistentPersonRepository(dataSource);
```
Comment (tsony-tsonev) [inline test/com/clouway/jdbc/task2/PersistentPersonRepositoryTest.java]: this should be one line

File: test/com/clouway/jdbc/task2/PersistentPersonRepositoryTest.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+        userRepository.register(123456789,"ivan",23,"dasas@abv.bg");
+        List<Person> users = userRepository.allPeopleInRepository();
+
+        assertThat(users.size(), is(equalTo(1)));
+    }
+
+    @Test
+    public void updateTest() {
```
Comment (tsony-tsonev) [inline test/com/clouway/jdbc/task2/PersistentPersonRepositoryTest.java]: name should be updateUser, also you should have at least one more test with more then one person cause now you might be updating only the first or the last person and not the one you need.

File: src/com/clouway/task2/domain/PersonRepository.java (hunk 1 above)
Comment (slavidichkov) [inline src/com/clouway/task2/domain/PersonRepository.java]: fixed

File: test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+
+        persistentTripRepository.deleteRepositoryContent();
+
+        assertThat(tripTableSize(dataSourceTrip), is(equalTo(0)));
+    }
+
+    @After
+    public void cleanUp() {
+//        new DatabaseCleaner(dataSourceTrip, "trip").cleanUp();
+//        new DatabaseCleaner(dataSourcePeople, "people").cleanUp();
```
Comment (tsony-tsonev) [inline test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java]: commented code

File: src/com/clouway/task2/domain/TripRepository.java (hunk 5 above)
Comment (slavidichkov) [inline src/com/clouway/task2/domain/TripRepository.java]: done

File: test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+    }
+
+    private int tripTableSize(DataSource dataSource) {
```
Comment (tsony-tsonev) [inline test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java]: numberOfTrips would make tests much more readable

```java
assertThat(numberOfTrips(dataSourceTrip), is(equalTo(0)));
```

also not sure you need to pass the datasource to this helper method cause you use it only for the trip dataset

File: src/com/clouway/task2/domain/Person.java (hunk 2 above)
Comment (slavidichkov) [inline src/com/clouway/task2/domain/Person.java]: fix

File: src/com/clouway/task2/domain/Trip.java (hunk 3 above)
Comment (slavidichkov) [inline src/com/clouway/task2/domain/Trip.java]: fixed

File: src/com/clouway/task2/domain/PersonRepository.java [hunk 11]
```diff
@@ -0,0 +1,12 @@
+package com.clouway.task2.domain;
+
+import java.util.List;
+
+/**
+ * @author Slavi Dichkov (slavidichkof@gmail.com)
+ */
+public interface PersonRepository {
+    void register(Person person);
+    void updatePerson(Person peerson);
```
Comment (tsony-tsonev) [inline src/com/clouway/task2/domain/PersonRepository.java]: update is enough  cause now you pass person

File: test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java (hunk 9 above)
Comment (slavidichkov) [inline test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java]: I uncommented it

File: src/com/clouway/task2/domain/TripRequest.java [hunk 12]
```diff
@@ -0,0 +1,22 @@
+package com.clouway.task2.domain;
+
+/**
+ * @author Slavi Dichkov (slavidichkof@gmail.com)
+ */
+public class TripRequest {
+    private final Person person;
+    private final Trip trip;
```
Comment (tsony-tsonev) [inline src/com/clouway/task2/domain/TripRequest.java]: also public final

File: test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java [hunk 13]
```diff
@@ -0,0 +1,206 @@
...
+        dataSourcePeople.setUser("root");
+        dataSourcePeople.setPassword("clouway.com");
+
+        new DatabaseCleaner(dataSourcePeople, "people").cleanUp();
+        new DatabaseCleaner(dataSourceTrip, "trip").cleanUp();
+    }
+
+    @Test
+    public void emptyRepository() {
+        assertThat(tripTableSize(dataSourceTrip), is(equalTo(0)));
```
Comment (tsony-tsonev) [inline test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java]: this test is not need cause it tests only the helper method which is not part of the trip repo interface

File: test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java (hunk 10 above)
Comment (slavidichkov) [inline test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java]: data source parameter was unnecessary and I changed name to numberOfTrips()

File: test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java (hunk 9 above)
Comment (tsony-tsonev) [inline test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java]: do you need it I see you make clean up in @Before method

File: file [hunk 14]
```diff
@@ -0,0 +1,2 @@
+slavi
+dichkov
```
Comment (tsony-tsonev) [inline file]: is this file needed

File: test/com/clouway/jdbc/task2/util/DatabaseCleaner.java [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+ * @author Slavi Dichkov (slavidichkof@gmail.com)
+ */
+public class DatabaseCleaner {
+
+    private final DataSource dataSource;
+    private final String[] tables;
+
+    public DatabaseCleaner(DataSource dataSource, String... users) {
+        this.dataSource = dataSource;
+        this.tables = users;
```
Comment (tsony-tsonev) [inline test/com/clouway/jdbc/task2/util/DatabaseCleaner.java]: this.tablse = tables;

File: file (hunk 14 above)
Comment (slavidichkov) [inline file]: No. I deleted it.

File: test/com/clouway/jdbc/task2/PersistentPersonRepositoryTest.java (hunk 6 above)
Comment (slavidichkov) [inline test/com/clouway/jdbc/task2/PersistentPersonRepositoryTest.java]: fixed

File: test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java [hunk 16]
```diff
@@ -0,0 +1,205 @@
...
+    }
+
+    @Test
+    public void emptyRepository() {
+        assertThat(numberOfTrips(), is(equalTo(0)));
+    }
+
+    @Test
+    public void oneTripRegisteredInRepository() {
+        TripRequest tripRequest1 = new TripRequest(new Person(123456789, "ivan", 23, "dasas@abv.bg"), new Trip("sliven", january(2010, 11), january(2010, 18)));
```
Comment (tsony-tsonev) [inline test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java]: do not add numbers if you use just one object

File: test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java [hunk 17]
```diff
@@ -0,0 +1,205 @@
...
+
+
+        PersistentTripRepository persistentTripRepository = new PersistentTripRepository(dataSourceTrip);
+        persistentTripRepository.register(tripRequest1);
+
+        assertThat(numberOfTrips(), is(equalTo(1)));
```
Comment (tsony-tsonev) [inline test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java]: you should also assert that the registered trip is the same one as the stored one

File: test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java [hunk 18]
```diff
@@ -0,0 +1,205 @@
...
+        persistentTripRepository.register(tripRequest4);
+        persistentTripRepository.register(tripRequest5);
+
+        List<Person> personList = persistentTripRepository.getPeopleInCityOnDate("sliven", january(2010, 13),january(2010, 17));
+
+        assertThat(personList.size(), is(equalTo(3)));
+    }
+
+    @Test
+    public void anotherTest() {
```
Comment (tsony-tsonev) [inline test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java]: very odd test name also nothing is asserted/tested

File: src/com/clouway/task2/domain/TripRequest.java (hunk 12 above)
Comment (slavidichkov) [inline src/com/clouway/task2/domain/TripRequest.java]: it's class is unnecessery

File: test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java [hunk 19]
```diff
@@ -0,0 +1,205 @@
...
+        persistentTripRepository.register(tripRequest4);
+
+
+        List<String> mostVisitedCities = persistentTripRepository.mostVisitedCities();
+
+        assertThat(mostVisitedCities.get(0), is("sliven"));
+    }
+
+    @Test
+    public void mostVisitedCities1() {
```
Comment (tsony-tsonev) [inline test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java]: another odd test name

File: test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java (hunk 13 above)
Comment (slavidichkov) [inline test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java]: fixed

File: test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java (hunk 18 above)
Comment (slavidichkov) [inline test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java]: I delete it

File: test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java (hunk 19 above)
Comment (slavidichkov) [inline test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java]: fixed

File: test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java (hunk 16 above)
Comment (slavidichkov) [inline test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java]: fixed

File: test/com/clouway/jdbc/task2/util/DatabaseCleaner.java (hunk 15 above)
Comment (slavidichkov) [inline test/com/clouway/jdbc/task2/util/DatabaseCleaner.java]: fixed

File: src/com/clouway/task2/domain/PersonRepository.java (hunk 11 above)
Comment (slavidichkov) [inline src/com/clouway/task2/domain/PersonRepository.java]: fixed

File: src/com/clouway/task2/adapter/PersistentPersonRepository.java (hunk 4 above)
Comment (slavidichkov) [inline src/com/clouway/task2/adapter/PersistentPersonRepository.java]: I understand but at the moment I will leave it this way

File: test/com/clouway/jdbc/task2/PersistentPersonRepositoryTest.java (hunk 7 above)
Comment (slavidichkov) [inline test/com/clouway/jdbc/task2/PersistentPersonRepositoryTest.java]: fixed

File: test/com/clouway/jdbc/task2/PersistentPersonRepositoryTest.java (hunk 8 above)
Comment (slavidichkov) [inline test/com/clouway/jdbc/task2/PersistentPersonRepositoryTest.java]: name is fixed

File: test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java (hunk 17 above)
Comment (slavidichkov) [inline test/com/clouway/jdbc/task2/PersistentTripRepositoryTest.java]: done

**Comment by tsony-tsonev** [comment]
You can merge this PR. LGTM , just remove trips.xml file cause i think is not needed.
